x-pack/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -98,3 +198,262 @@ async function indexAlerts(
     await client.bulk({ body, refresh: true });
   }
 }
+
+const createPolicy = async (
+  kbnClient: KbnClient,
+  policyName: string,
+  endpointPackageVersion: string
+): Promise<CreatePackagePolicyResponse['item']> => {
+  // Create Agent Policy first
+  const newAgentPolicyData: CreateAgentPolicyRequest['body'] = {
+    name: `Policy for ${policyName}`,
+    description: '',
+    namespace: 'default',
+  };
+  let agentPolicy;
+  try {
+    agentPolicy = (await kbnClient.request({
+      path: AGENT_POLICY_API_ROUTES.CREATE_PATTERN,
+      method: 'POST',
+      body: newAgentPolicyData,
+    })) as AxiosResponse<CreateAgentPolicyResponse>;
+  } catch (error) {
+    throw new Error(`create policy ${error}`);
+  }
+
+  // Create Package Configuration
+  const newPackagePolicyData: CreatePackagePolicyRequest['body'] = {
+    name: policyName,
+    description: 'Protect the worlds data',
+    policy_id: agentPolicy.data.item.id,
+    enabled: true,
+    output_id: '',
+    inputs: [
+      {
+        type: 'endpoint',
+        enabled: true,
+        streams: [],
+        config: {
+          policy: {
+            value: policyConfigFactory(),
+          },
+        },
+      },
+    ],
+    namespace: 'default',
+    package: {
+      name: 'endpoint',
+      title: 'endpoint',
+      version: endpointPackageVersion,
+    },
+  };
+  const packagePolicy = (await kbnClient.request({
+    path: PACKAGE_POLICY_API_ROUTES.CREATE_PATTERN,
+    method: 'POST',
+    body: newPackagePolicyData,
+  })) as AxiosResponse<CreatePackagePolicyResponse>;
+  return packagePolicy.data.item;
+};
+
+const getEndpointPackageInfo = async (
+  kbnClient: KbnClient
+): Promise<GetPackagesResponse['response'][0]> => {
+  const endpointPackage = ((await kbnClient.request({
+    path: `${EPM_API_ROUTES.LIST_PATTERN}?category=security`,
+    method: 'GET',
+  })) as AxiosResponse<GetPackagesResponse>).data.response.find(
+    (epmPackage) => epmPackage.name === 'endpoint'
+  );
+
+  if (!endpointPackage) {
+    throw new Error('EPM Endpoint package was not found!');
+  }
+
+  return endpointPackage;
+};
+
+const fleetEnrollAgentForHost = async (
+  kbnClient: KbnClient & {
+    requestWithApiKey: (path: string, init?: RequestInit | undefined) => Promise<Response>;
+  },
+  endpointHost: HostMetadata,
+  agentPolicyId: string
+): Promise<undefined | PostAgentEnrollResponse['item']> => {
+  // Get Enrollement key for host's applied policy
+  const enrollmentApiKey = await kbnClient
```

Comment:
Could consider making this `request.then.then.catch` its own function called `getEnrollmentApiKey` or similar